### PR TITLE
INFO message when removing duplicate products should give the correct number of products based on filters

### PR DESCRIPTION
**Pull request by snbianco**
Related to #2496 and #2497

Quick fix so that the number of products given in the INFO message in `_remove_duplicate_products` reflects any filters that were applied to the products. I did this by switching the order of the calls in `download_products` so that filters are applied to the product list before duplicates are removed.

### CODE REVIEW

File: astroquery/mast/observations.py
Comment (bsipocz) [inline astroquery/mast/observations.py:747]: Would this cause some performance degradation as now the duplicates end up going into `filter_products()`? Does filtering or duplicate removal is more costly, or it doesn't matter in practice?

Comment (snbianco) [reply astroquery/mast/observations.py:747]: Hmm, I guess it would depend on the particular query. You may have a good amount of duplicate products, but if you set strict filters (like `mrp_only = True` in some cases), it would be better to filter first. The time complexity of either filtering or removing duplicates would be O(n). I don't think that it matters too much in practice.

Comment (bsipocz) [reply astroquery/mast/observations.py:747]: OK, then I will leave this to your judgment as the maintainer of this module, and merge in as is.

Thanks!

**Review by bsipocz — APPROVED** [review]

**Pull request merged**
